src/fsharp/FSharp.Core/string.fs

```diff
@@ -43,13 +43,24 @@ namespace Microsoft.FSharp.Core
 
         [<CompiledName("MapIndexed")>]
         let mapi (mapping: int -> char -> char) (str:string) =
-            if String.IsNullOrEmpty str then
+            let len = length str
+            if len = 0 then 
                 String.Empty
             else
-                let res = StringBuilder str.Length
-                let f = OptimizedClosures.FSharpFunc<_,_,_>.Adapt(mapping)
-                str |> iteri (fun i c -> res.Append(f.Invoke(i, c)) |> ignore)
-                res.ToString()
+                let result = str.ToCharArray()
+                let f = OptimizedClosures.FSharpFunc<_,_,_>.Adapt mapping
```

Comment:
nit: the change to remove the parentheses isn't necessary here

Reply:
@cartermp Thanks, fixed here: https://github.com/dotnet/fsharp/pull/9482/commits/3744f5301c08149f3e1ed5fc8bc159c124a72898 (which also fixes the CI errors).

I'm no fan of parens, hence I removed it ;)